### PR TITLE
fix(reports) support functions as values in ping reports

### PR DESCRIPTION
**Pull request by kinman**
If a ping value is a function, execute the function and then evaluate the value. 
This allows functions to return tables that will be JSON-encoded before being
added to the ping message.

### CODE REVIEW

**Comment by kinman** [comment]
I'm ambivalent about the label of the change, will change to `feat` if that's more agreeable. While the author of a function could JSON-encode themselves, that's exposing report internals -- serialization choices -- that, if changed, would require changes in the calling code. Better to keep report data and report formatting responsibilities separate.

**Comment by thibaultcha** [comment]
That's fine and understandable - what I meant was with regards to the urgency of the feature being merged, which seems rather low for such a private API. Still willing to merge it if it comes with tests (especially since the goal is to provide a stronger encapsulation)!

**Comment by kinman** [comment]
Added test coverage for serializing tables as JSON.

File: spec/01-unit/013-reports_spec.lua [hunk 1]
```diff
@@ -28,6 +30,8 @@ describe("reports", function()
       assert.matches("hello=world", res, nil, true)
       assert.matches("signal=stub", res, nil, true)
       assert.matches("baz=bat", res, nil, true)
+      assert.matches("foobar=" .. cjson.encode({ foo = "bar" }), res, nil, true)
+      assert.matches("bazbat=" .. cjson.encode({ baz = "bat" }), res, nil, true)
```
Comment (thibaultcha) [inline spec/01-unit/013-reports_spec.lua]: It seems like `cjson` is `nil` in this test, causing the CI to fail.

Comment (kinman) [reply spec/01-unit/013-reports_spec.lua]: Yes. Bad monkey-patching. Passing now.

**Comment by thibaultcha** [comment]
Manually merged to use `cjson` instead of `cjson.safe`; closing this!

**Pull request closed without merging**